### PR TITLE
Delete all private groups

### PR DESCRIPTION
**Pull request by sipec**
I've already N/A'ed all remaining multi, canceled all private bounties, and closed all private polls. Also, everyone who asked to keep their private markets / groups has already had their markets turned unlisted and their group turned curated.
so running this script should just take care of private groups once and for all

- comments are deleted
- remaining markets get N/A'ed
- contracts and answers get redacted
- private groups are literally deleted

### CODE REVIEW

File: backend/scripts/delete-private-groups.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  )
+
+  for (const contract of unresolvedContracts) {
+    const creator = await getUser(contract.creatorId)
+    if (!creator) {
+      console.error(`Creator not found for contract ${contract.id}`)
+      continue
+    }
+    await resolveMarketHelper(contract, sinclair, creator, {
+      outcome: 'CANCELLED',
```
Comment (jahooma) [inline backend/scripts/delete-private-groups.ts:34]: Wasn't it "CANCEL"?

File: backend/scripts/delete-private-groups.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    || jsonb_build_object('slug', 'private-auto-deleted-' || random_alphanumeric(8))
+    || jsonb_build_object('lastUpdatedTime', ts_to_millis(now()))
+    where visibility = 'private'
+    and (deleted != true or resolution is null)`
+  )
+  // note that answers will get overwritten by the denormalizer, but we set to empty to prevent data leakage.
+
+  console.log('deleting all private groups')
+  // copied from scripts/delete-group.ts
+  const groups = pg.many(
```
Comment (jahooma) [inline backend/scripts/delete-private-groups.ts:72]: nit: I would name this variable groupIds

File: backend/scripts/delete-private-groups.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  console.log(`overwriting and marking deleted all private contracts`)
+  await pg.none(
+    `update contracts
+    set data = data || '{
+        "wasPrivate": true,
+        "visibility": "unlisted",
+        "isRanked": false,
+        "deleted": true
+        "question": "Private Question (deleted)",
```
Comment (jahooma) [inline backend/scripts/delete-private-groups.ts:57]: I think we should ovewrite the slug as well

Comment (jahooma) [reply backend/scripts/delete-private-groups.ts:57]: Nvm I see you did

**Review by jahooma — APPROVED** [review]
Thanks Sinclair, looks good!

Was there anything in the notifications that we wanted to scrub? Might be worth removing the contract question if that was in plain text in a notification.

File: backend/scripts/delete-private-groups.ts (hunk 3 above)
Comment (sipec) [reply backend/scripts/delete-private-groups.ts:57]: it is overwritten on line 63

**Pull request merged**